### PR TITLE
Add spec text for extend operators

### PR DESCRIPTION
**Pull request by binji**
See published version here: https://webassembly.github.io/threads

### CODE REVIEW

File: document/core/util/math.def [hunk 1]
```diff
@@ -735,6 +735,9 @@
 .. |extend| mathdef:: \xref{exec/numerics}{op-extend_u}{\F{extend}}
 .. |extendu| mathdef:: \xref{exec/numerics}{op-extend_u}{\F{extend\_u}}
 .. |extends| mathdef:: \xref{exec/numerics}{op-extend_s}{\F{extend\_s}}
+.. |extend8s| mathdef:: \xref{exec/numerics}{op-extend8_s}{\F{extend8\_s}}
```
Comment (rossberg-chromium) [inline document/core/util/math.def]: These should also be named iextend for consistency (and nit, be moved up to the other i* macros).

File: document/core/syntax/instructions.rst [hunk 2]
```diff
@@ -151,7 +154,12 @@ Occasionally, it is convenient to group operators together according to the foll
 
 .. math::
    \begin{array}{llll}
-   \production{unary operator} & \unop &::=& \iunop ~|~ \funop \\
+   \production{unary operator} & \unop &::=&
+     \iunop ~|~
+     \funop ~|~
+     \EXTEND\K{8\_s} ~|~
```
Comment (binji) [inline document/core/syntax/instructions.rst]: Oh, good idea.

**Comment by binji** [comment]
OK, fixed. Gonna land assuming I addressed these issues, please tell me if I missed anything!

File: document/core/util/math.def [hunk 3]
```diff
@@ -710,6 +710,7 @@
 .. |iles| mathdef:: \xref{exec/numerics}{op-ile_s}{\F{ile\_s}}
 .. |igeu| mathdef:: \xref{exec/numerics}{op-ige_u}{\F{ige\_u}}
 .. |iges| mathdef:: \xref{exec/numerics}{op-ige_s}{\F{ige\_s}}
+.. |iextends| mathdef:: \xref{exec/numerics}{op-iextend_s}{\F{iextend{M}\_s}}
```
Comment (rossberg-chromium) [inline document/core/util/math.def]: Minor issue: to get the proper type setting this needs to be defined as {\F[iextend}M\F{\_s}}.
And for consistency, can we name the macro iextendns and the label op-iextendn_s?.

**Pull request merged**